mono/metadata/icall.c

```diff
@@ -875,6 +875,18 @@ ves_icall_System_Array_CreateInstanceImpl (MonoReflectionTypeHandle type, MonoAr
 	return mono_array_new_full_handle (MONO_HANDLE_DOMAIN (type), aklass, sizes, lower_bounds, error);
 }
 
+#ifdef ENABLE_NETCORE
+void
+ves_icall_System_Array_GetRawArrayGeometry(MonoArrayHandle arr, gint32* numComponents, gint32* elementSize, gint32* lowerBound, gint32* containsGCPointers, MonoError* error)
+{
+	MonoClass* arr_klass = mono_handle_class (arr);
+	*elementSize = mono_array_element_size (arr_klass);
+	*lowerBound = MONO_HANDLE_GETVAL (arr, bounds) ? MONO_HANDLE_GETVAL (arr, bounds [0].lower_bound) : 0;
+	*containsGCPointers = m_class_has_references (m_class_get_element_class (arr_klass));
+	*numComponents = MONO_HANDLE_GETVAL (arr, max_length);
```

Comment:
I am looking into some intrinsics for `Array` and it makes me wonder if we can accomplish this without a new icall. `GetElementSize` is currently not optimized but I plan to do it. `GetLowerBound` is already intrinsic, the codegen is not perfect for constants but that's totally fixable. `numComponents` is already available through the raw array data on the managed side IIRC. That leaves out only `containsGCPointers`.